tests/e2e/sample-applications/mongo-persistent/mongo-persistent-block.yaml

```diff
@@ -0,0 +1,260 @@
+    supplementalGroups:
+      type: RunAsAny
+    volumes:
+    - '*'
+    users:
+    - system:admin
+    - system:serviceaccount:mongo-persistent:mongo-persistent-sa
+  - kind: ImageStream
+    apiVersion: image.openshift.io/v1
+    metadata:
+      name: todolist-mongo-go
+      namespace: mongo-persistent
+      labels:
+        app: todolist
+    spec: {}
+    status:
+      dockerImageRepository: ''
+  - apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      annotations:
+        template.alpha.openshift.io/wait-for-ready: 'true'
+      name: mongo
+      namespace: mongo-persistent
+      labels:
+        e2e-app: "true"
+    spec:
+      selector:
+        matchLabels:
+          app: mongo
+      strategy:
+        type: Recreate
+      template:
+        metadata:
+          labels:
+            e2e-app: "true"
+            app: mongo
+        spec:
+          serviceAccountName: mongo-persistent-sa
+          securityContext:
+            runAsUser: 0  
+          # Used to format the block device also touch one file which is used to determine when the
+          # block device was formatted. We use the same image as 
+          initContainers:
+            - image: mongo:latest
+              securityContext:
+                privileged: true
+              name: setup-block-device
+              command:
+                - "sh"
+                - "-c"
+                - |
+                  DEVICE="/dev/xvdx"
+                  MOUNT_POINT="/data/xvdx"
+                  if [ ! -e $DEVICE ]; then
+                    echo "$DEVICE does not exist."
+                    exit 1
+                  fi
+                  echo "Formatting $DEVICE"
+                  dnf install -y e2fsprogs
+                  mkfs.ext4 $DEVICE
+                  mkdir -p $MOUNT_POINT
+                  echo "Mounting $DEVICE in the $MOUNT_POINT"
+                  mount $DEVICE $MOUNT_POINT
+                  echo $(date +%s) > $MOUNT_POINT/format_timestamp
+                  echo "Recorded timestamp: $(cat $MOUNT_POINT/format_timestamp)"                  
+                  umount $MOUNT_POINT              
+              volumeDevices:
+                - name: block-volume-pv
+                  devicePath: /dev/xvdx
+          containers:
+            - image: mongo:latest
+              name: mongo
+              securityContext:
+                privileged: true
+              env:
+                - name: MONGO_INITDB_ROOT_USERNAME
+                  value: changeme
+                - name: MONGO_INITDB_ROOT_PASSWORD
+                  value: changeme
+                - name: MONGO_INITDB_DATABASE
+                  value: todolist
+              ports:
+                - containerPort: 27017
+                  name: mongo
+              resources:
+                limits:
+                  memory: 512Mi
+              command: ["sh", "-c", "mkdir -p /data/xvdx && mount /dev/xvdx /data/xvdx && mongod --bind_ip_all --dbpath /data/xvdx"]
```

Comment:
so `args` should be `mongod --bind_ip_all --dbpath /data/xvdx` https://hub.docker.com/layers/library/mongo/latest/images/sha256-075e0577e2989efee25f8e6cd615ae1ce84da1f8c79adc3557aaf253dbf7a5e0?context=explore

can `mkdir -p /data/xvdx && mount /dev/xvdx /data/xvdx` be run in initcontainer?

Reply:
No it can't be as it's not persistent across restarts.

Reply:
change command (container entrypoint) to `sh` and args (container command) to `docker-entrypoint.sh && mkdir -p /data/xvdx && mount /dev/xvdx /data/xvdx && mongod --bind_ip_all --dbpath /data/xvdx`?

Reply:
@mateusoliveira43 thanks mate! That did the trick, small update to the command but overall the idea of passing it as the last arg worked.

Reply:
This is the command that worked:

```yaml
              command:
                - "sh"
                - "-c"
                - |
                  DEVICE="/dev/xvdx"
                  MOUNT_POINT="/data/db"
                  mkdir -p $MOUNT_POINT
                  mount $DEVICE $MOUNT_POINT
                  docker-entrypoint.sh mongod --auth --bind_ip_all --dbpath $MOUNT_POINT
```